Source/WebCore/html/canvas/WebGL2RenderingContext.cpp

```diff
@@ -1721,8 +1719,10 @@ std::optional<WebGL2RenderingContext::ActiveQueryKey> WebGL2RenderingContext::va
 
 void WebGL2RenderingContext::beginQuery(GCGLenum target, WebGLQuery& query)
 {
+    if (isContextLost())
+        return;
```

Comment:
Instead of writing `if (isContextLost()) return;` in all functions, should this check be performed at the JS binding level and the binding generator write the code for us? Then, we'd never forgot to write the check.

Reply:
Yeah, that would be a good idea perhaps. There's couple of cases that need to be handled manually -- these are in fact probably marked by WebGL spec IDL as WebGLHandlesContextLoss or similar annotation.